### PR TITLE
Add libsodium

### PR DESCRIPTION
**Pull request by Jean85**
As reported in travis-ci/travis-ci#8863, PHP 7.2 is missing libsodium and Argon2i support; this is wrong because by default PHP 7.2 comes with that, so a lot of builds are failing or required to use a long workaround and install it manually.

The original commit isn't mine, but it should work to enable it from PHP 7.2 and forward.

This PR superseedes #19.

### CODE REVIEW

**Comment by tristanbes** [comment]
When do we want it ? nooooow :D

File: bin/install-password-argon2 [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -o xtrace
+set -o errexit
+
+# If PHP < 7.2, exit
+if [[ $VERSION =~ ^master$ ]]; then
+	echo 'unknown PHP version; skip password-argon2 steps'
+	exit 0
+elif [ "$(printf "7.2\n$VERSION" | sort -V | head -n1)" != "7.2" ]; then
```
Comment (BanzaiMan) [inline bin/install-password-argon2]: We have a helper function for comparing software versions (`vers2int`). Please use that.

Comment (BanzaiMan) [reply bin/install-password-argon2]: Regrettably, 7.2 is no longer the only valid value you'd want to accept, so this condition needs an update.

**Review by BanzaiMan — COMMENTED** [review]
Suggestions

File: bin/install-password-argon2 (hunk 1 above)
Comment (Jean85) [reply bin/install-password-argon2]: Well, this PR is 4 months old...

Do you have any example of this `vers2int` to follow?

Comment (BanzaiMan) [reply bin/install-password-argon2]: Sorry about the digression there. This repo does not have the helper function. I was thinking of another repo.

Comment (Jean85) [reply bin/install-password-argon2]: Oh ok! So is this mergeable as-is?

Comment (BanzaiMan) [reply bin/install-password-argon2]: Not quite. It'll now need to consider `7.3*` (and potentially `7.4*`). It is probably easier to test for `<7.2`.

**Comment by Jean85** [comment]
@BanzaiMan fixed!

File: bin/install-password-argon2 [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -o xtrace
+set -o errexit
+
+# If PHP < 7.2, exit
+if [[ $VERSION =~ ^master$ ]]; then
+	echo 'unknown PHP version; skip password-argon2 steps'
+	exit 0
+elif [ "$(printf "7.2\n$VERSION" | sort -V | head -n1)" < "7.2" ]; then
```
Comment (BanzaiMan) [inline bin/install-password-argon2]: Does this `elif` statement work on your machine?

```sh-session
bash-4.4$ VERSION=foo; if [ $(printf "7.2\n$VERSION" | sort -V | head -n1) < "7.2" ]; then echo skip; fi
bash: 7.2: No such file or directory
bash-4.4$ VERSION=7.1; if [ $(printf "7.2\n$VERSION" | sort -V | head -n1) < "7.2" ]; then echo skip; fi
bash: 7.2: No such file or directory
```

It seems to me that `<` is interpreted as redirection, and bash is looking for a file named `7.2` instead.

Perhaps you want `[[]]`:

```sh-session
bash-4.4$ VERSION=7.2; if [[ "$(printf "7.2\n$VERSION" | sort -V | head -n1)" < "7.2" ]]; then echo skip; else echo go; fi
go
bash-4.4$ VERSION=7.1; if [[ "$(printf "7.2\n$VERSION" | sort -V | head -n1)" < "7.2" ]]; then echo skip; else echo go; fi
skip
bash-4.4$ VERSION=10; if [[ "$(printf "7.2\n$VERSION" | sort -V | head -n1)" < "7.2" ]]; then echo skip; else echo go; fi
go
bash-4.4$ VERSION=foo; if [[ "$(printf "7.2\n$VERSION" | sort -V | head -n1)" < "7.2" ]]; then echo skip; else echo go; fi
go
```

Comment (Jean85) [reply bin/install-password-argon2]: Nope, you're right! Fixing!

**Comment by Majkl578** [comment]
While this PR adds only and ONLY libsodium, the shell file is incorrectly named _install-password-argon2_ which it has nothing to do with. Hence `--with-password-argon2` is wrong too.
Spotted by @cmb69: https://github.com/travis-ci/travis-ci/issues/9717#issuecomment-426698254

File: bin/install-password-argon2 [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -o xtrace
+set -o errexit
+
+# If PHP < 7.2, exit
+if [[ $VERSION =~ ^master$ ]]; then
```
Comment (Majkl578) [inline bin/install-password-argon2:7]: @Jean85 Can you explain this condition? It doesn't make sense to me, master (aka nightly) > 7.2, it should build for master too. Thanks.

Comment (Jean85) [reply bin/install-password-argon2:7]: I'm sorry, the original code wasn't mine (see #19) and it predated 7.2, so it may be incorrect right now...

Comment (Majkl578) [reply bin/install-password-argon2:7]: Np, see #22 + #23. :)

**Comment by Majkl578** [comment]
@Jean85 #22 got merged, if you can let me know how it works. :+1:

**Pull request merged**